packages/ods/src/components/accordion/src/components/ods-accordion/ods-accordion.tsx

```diff
@@ -0,0 +1,95 @@
+import type { FunctionalComponent } from '@stencil/core';
+import { Component, Event, type EventEmitter, Host, Prop, Watch, h } from '@stencil/core';
+import { ODS_ICON_NAME } from '../../../../icon/src';
+
+@Component({
+  shadow: true,
+  styleUrl: 'ods-accordion.scss',
+  tag: 'ods-accordion',
+})
+export class OdsAccordion {
+  private detailsElement?: HTMLDetailsElement;
+
+  @Prop({ reflect: true }) public isDisabled: boolean = false;
+  @Prop({ mutable: true, reflect: true }) public isOpen: boolean = false;
+
+  @Event() odsAccordionToggle!: EventEmitter<boolean>;
+
+  handleToggle(event: Event): void {
+    if (this.isDisabled) {
+      return event.preventDefault();
+    }
+
+    if (this.isOpen) {
+      this.isOpen = false;
+    } else {
+      this.isOpen = true;
+    }
+
+    this.odsAccordionToggle.emit();
+  }
+
+  @Watch('isOpen')
+  onOpenChange(): void{
+    if (this.isOpen && !this.isDisabled) {
+      this.detailsElement?.setAttribute('open', '');
+    } else {
+      this.detailsElement?.removeAttribute('open');
+    }
+  }
+
+  componentDidLoad(): void {
+    this.onOpenChange();
+  }
+
+  handleClick(event: MouseEvent): void {
+    this.handleToggle(event);
+  }
+
+  handleKeyDown(event: KeyboardEvent): void {
+    if (event.key === ' ' || event.key === 'Enter') {
+      this.handleToggle(event);
+    }
+  }
+
+  render(): FunctionalComponent {
+    return (
+      <Host class='ods-accordion'>
+        <details
+          class={{
+            'ods-accordion__wrapper': true,
+            'ods-accordion__wrapper--is-disabled': this.isDisabled,
+            'ods-accordion__wrapper--is-open': this.isOpen,
+          }}
+          tabIndex={this.isDisabled ? -1 : 0}
+          onClick={(event) => this.handleClick(event)}
+          onKeyDown={(event) => this.handleKeyDown(event)}
```

Comment:
You'll have to preventDefault onKeyDown for space, to prevent page scroll (Firefox), see select or tag